language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/semantictokens/SemanticTokensVisitor.java

```diff
@@ -293,6 +304,7 @@ public void visit(TypeDefinitionNode typeDefinitionNode) {
                     }
                     if (left == SyntaxKind.READONLY_TYPE_DESC || right == SyntaxKind.READONLY_TYPE_DESC) {
                         modifiers = TokenTypeModifiers.DECLARATION.getId() | TokenTypeModifiers.READONLY.getId();
+                        refModifiers = TokenTypeModifiers.READONLY.getId();
```

Comment:
We have enclosed a considerable chunk of if-clause within a case. Try to move this logic to a separate method, if possible

Reply:
Let's keep this since there are multiple variables assigned inside the logic.

Reply:
+1. Let's try to refactor this in a later phase since this logic seems to grow and having issues with the readability.